src/sknnr/_gnn.py

```diff
@@ -1,16 +1,14 @@
-from ._base import _TransformedKNeighborsRegressor
+from ._base import TransformedKNeighborsRegressor, YFitMixin
 from .transformers import CCATransformer
 
 
-class GNNRegressor(_TransformedKNeighborsRegressor):
+class GNNRegressor(YFitMixin, TransformedKNeighborsRegressor):
     def __init__(self, n_components=None, **kwargs):
         super().__init__(**kwargs)
         self.n_components = n_components
 
-    def fit(self, X, y, y_fit=None):
-        y_fit = y_fit if y_fit is not None else y
-        self.transform_ = CCATransformer(self.n_components).fit(X, y=y_fit)
-        return super().fit(X, y)
+    def _get_transformer(self):
```

Comment:
Should we be providing type information here as you did for `EuclideanKNNRegressor` (i.e. `TransformerMixin`).  Also on `MahalanobisKNNRegressor` and `MSNRegressor`

Reply:
Definitely, good call!